components/style/properties/gecko.mako.rs

```diff
@@ -1274,6 +1274,140 @@ fn static_assert() {
 
     <% impl_color("stop_color", "mStopColor") %>
 
+    pub fn set_clip_path(&mut self, v: longhands::clip_path::computed_value::T) {
+        use gecko_bindings::bindings::{Gecko_NewBasicShape, Gecko_ReleaseClipPath};
+        use gecko_bindings::structs::StyleClipPathGeometryBox;
+        use gecko_bindings::structs::{StyleBasicShape, StyleBasicShapeType, StyleShapeSourceType};
+        use gecko_bindings::structs::{StyleClipPath, StyleFillRule};
+        use gecko_conversions::basic_shape::set_corners_from_radius;
+        use gecko_values::GeckoStyleCoordConvertible;
+        use values::computed::basic_shape::*;
+        let ref mut clip_path = self.gecko.mClipPath;
+        // clean up existing struct
+        unsafe { Gecko_ReleaseClipPath(clip_path) };
+
+        clip_path.mType = StyleShapeSourceType::None_;
+
+        match v.0 {
+            ShapeSource::Url(..) => println!("stylo: clip-path: url() not yet implemented"),
+            ShapeSource::None => {} // don't change the type
+            ShapeSource::Box(reference) => {
+                clip_path.mReferenceBox = reference.into();
+                clip_path.mType = StyleShapeSourceType::Box;
+            }
+            ShapeSource::Shape(servo_shape, maybe_box) => {
+
```

Comment:
Nit: the other match arms don't have a blank line at the start, so maybe drop this one?